daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -809,31 +805,24 @@ convertExpr env0 e = do
           TContractId{} -> asLet
           TUpdate{} -> asLet
           TScenario{} -> asLet
-          tcon0 -> do
-              ctor <- toCtor env con
+          tcon -> do
+              ctor@(Ctor _ fldNames fldTys) <- toCtor env con
               if not (isRecordCtor ctor)
-                then do
+                then convertLet env bind scrutinee $  \env -> do
+                  bind' <- convertExpr env (Var bind)
                   ty <- convertType env $ varType bind
                   alt' <- convertAlt env ty alt
-                  bind' <- convVarWithType env bind
-                  pure $
-                    ELet (Binding bind' scrutinee') $
-                    ECase (EVar $ convVar bind) [alt']
-                else mkProj ctor tcon0 scrutinee'
+                  pure $ ECase bind' [alt']
+                -- NOTE(MH): The 'zipExact' below fails for pattern matches on constructors with existentials.
```

Comment:
Comment seems to add zero value, since the same comment is in the error message

Reply:
You're right. I removed it.